### PR TITLE
feat: Add SARIF format generation for GHAS

### PR DESCRIPTION
**Pull request by ashujha301**
1. SARIF input added.
2. Binary version updated to 1.4.0.
3. @zeit/ncc updated to @vercel/ncc.
4. Updated readme.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,7 +35,7 @@
     "@types/jest": "^26.0.15",
     "@types/node": "^14.14.7",
     "@typescript-eslint/parser": "^4.8.0",
```
Comment (ginilpg) [inline package.json:37]: @ashujha301 Any reason to remove this?

File: src/index.ts [hunk 2]
```diff
@@ -1,16 +1,26 @@
 import * as core from '@actions/core';
-import {cicheck, upload, whoami} from './tool';
+import {cicheck, upload, whoami, cicheckWithSarif} from './tool';
 import {getInputs} from './input-helper';
 
 async function run(): Promise<void> {
   try {
+    core.debug('APPKNOX-GITHUB-ACTION');
```
Comment (ginilpg) [inline src/index.ts]: Remove debuggers

File: src/constants.ts [hunk 3]
```diff
@@ -26,4 +27,4 @@ export enum RiskThresholdOptions {
   CRITICAL = 'CRITICAL'
 }
 
-export const binaryVersion = '1.2.0';
+export const binaryVersion = '1.4.6';
```
Comment (ginilpg) [inline src/constants.ts]: any consequences if we upgraded the version here?

**Review by ginilpg — CHANGES_REQUESTED** [review]

File: src/appknox-inputs.ts [hunk 4]
```diff
@@ -15,4 +15,9 @@ export interface AppknoxInputs {
    * Minimum risk to fail CI
    */
   riskThreshold: RiskThresholdOptions;
+
+  /**
+   * Enable SARIF format
+   */
+  sarif: string;
```
Comment (cosmosgenius) [inline src/appknox-inputs.ts]: shouldn't this be boolean?

File: src/constants.ts [hunk 5]
```diff
@@ -1,7 +1,8 @@
 export enum Inputs {
   AppknoxAccessToken = 'appknox_access_token',
   Path = 'file_path',
-  RiskThreshold = 'risk_threshold'
+  RiskThreshold = 'risk_threshold',
+  sarif = 'sarif',
```
Comment (cosmosgenius) [inline src/constants.ts]:
```suggestion
  Sarif = 'sarif',
```

File: src/index.ts [hunk 6]
```diff
@@ -8,9 +8,17 @@ async function run(): Promise<void> {
...
+      await sarifReport(fileID);
+    }
     await cicheck(inputs.riskThreshold, fileID);
-  } catch (err) {
-    core.setFailed(err.message);
+  } catch (err: any) {
+    if (err instanceof Error) {
+      core.setFailed(err.message);
+    } else {
+      core.setFailed('Unknown error occurred');
```
Comment (cosmosgenius) [inline src/index.ts]: the msg should include the err object as well to know what error happened.

File: src/input-helper.ts [hunk 7]
```diff
@@ -10,6 +10,7 @@ export function getInputs(): AppknoxInputs {
     required: true
   });
   const path = core.getInput(Inputs.Path, {required: true});
+  const sarifBool = core.getInput(Inputs.sarif, {required: true});
```
Comment (cosmosgenius) [inline src/input-helper.ts]: This is not a required field right?.

**Review by cosmosgenius — CHANGES_REQUESTED** [review]

File: action.yml [hunk 8]
```diff
@@ -12,6 +12,10 @@ inputs:
     description: 'Minimum risk. Valid options are LOW, MEDIUM, HIGH, CRITICAL. Defaults to "LOW"'
     required: true
     default: 'LOW'
+  sarif:
+    description: 'Enable SARIF format report. Defaults to false'
+    required: true
+    default: 'false'
```
Comment (cosmosgenius) [inline action.yml]:
```suggestion
    default: 'disabled'
```

**Review by cosmosgenius — CHANGES_REQUESTED** [review]

**Review by cosmosgenius — APPROVED** [review]

**Pull request merged**